packages/runtime/container-runtime/src/gc/gcTelemetry.ts

```diff
@@ -119,76 +124,111 @@ export class GCTelemetryTracker {
 			return false;
 		}
 
+		// Non-tombstone events are logged once per event per node. A unique id is generated by joining
+		// node state (inactive / sweep ready), node's id and usage (loaded / changed / revived).
 		if (this.loggedUnreferencedEvents.has(uniqueEventId)) {
 			return false;
 		}
 		return true;
 	}
 
 	/**
-	 * Called when a node is used. If the node is not active, log an event indicating object is used when its not active.
+	 * Called when a node is used. If the node is not active or tombstone, log telemetry indicating object is used
+	 * when it should not have.
 	 */
 	public nodeUsed(nodeUsageProps: INodeUsageProps) {
 		// If there is no reference timestamp to work with, no ops have been processed after creation. If so, skip
 		// logging as nothing interesting would have happened worth logging.
-		// If the node is not unreferenced, skip logging.
-		const nodeStateTracker = this.getNodeStateTracker(nodeUsageProps.id);
-		if (!nodeStateTracker || nodeUsageProps.currentReferenceTimestampMs === undefined) {
+		if (nodeUsageProps.currentReferenceTimestampMs === undefined) {
 			return;
 		}
-
-		// We log these events once per event per node. A unique id is generated by joining node state (inactive / sweep ready),
-		// node's id and usage (loaded / changed / revived).
-		const uniqueEventId = `${nodeStateTracker.state}-${nodeUsageProps.id}-${nodeUsageProps.usageType}`;
-		const nodeType = this.getNodeType(nodeUsageProps.id);
-		if (
-			!this.shouldLogNonActiveEvent(
-				nodeUsageProps.id,
-				nodeType,
-				nodeUsageProps.usageType,
-				nodeStateTracker,
-				uniqueEventId,
-			)
-		) {
+		// If the node's unreferenced state is not tracked or it's not tombstoned, there is nothing to log.
+		const nodeStateTracker = this.getNodeStateTracker(nodeUsageProps.id);
+		if (nodeStateTracker === undefined && !nodeUsageProps.isTombstoned) {
 			return;
 		}
 
-		// Add the unique event id so that we don't generate a log for this event again in this session..
-		this.loggedUnreferencedEvents.add(uniqueEventId);
-
-		const state = nodeStateTracker.state;
+		const nodeType = this.getNodeType(nodeUsageProps.id);
 		const { usageType, currentReferenceTimestampMs, packagePath, id, fromId, ...propsToLog } =
 			nodeUsageProps;
-		const eventProps: Omit<IUnreferencedEventProps, "state" | "usageType"> = {
+		const tombstoneProps: Partial<IUnreferencedEventProps> = {
```

Comment:
Maybe call this something more generic since it's not just used for Tombstone.  And consider putting all the Tombstone stuff in a function, it'll make this function easier to read.  Maybe other parts of it get pulled into helpers too.

Reply:
Also, using type `Partial` casts doubt on whether all the properties are there.  I like the `Omit` one better.  I'm assuming there are other props besides `state` and `usageType` missing here?